addons/binding/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/telnet/DenonMarantzTelnetClient.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.denonmarantz.internal.telnet;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.denonmarantz.internal.config.DenonMarantzConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manage telnet connection to the Denon/Marantz Receiver
+ *
+ * @author Jeroen Idserda
+ * @author Jan-Willem Veldhuis
+ */
+public class DenonMarantzTelnetClient extends Thread {
```

Comment:
Please make it a `Runnable` and schedule it with the scheduler (`scheduler.submit`..)